core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -66,108 +69,169 @@ class LogManager(val logDirs: Array[File],
   private val logs = new Pool[TopicPartition, Log]()
   private val logsToBeDeleted = new LinkedBlockingQueue[Log]()
 
-  createAndValidateLogDirs(logDirs)
-  private val dirLocks = lockLogDirs(logDirs)
-  private val recoveryPointCheckpoints = logDirs.map(dir => (dir, new OffsetCheckpointFile(new File(dir, RecoveryPointCheckpointFile)))).toMap
-  private val logStartOffsetCheckpoints = logDirs.map(dir => (dir, new OffsetCheckpointFile(new File(dir, LogStartOffsetCheckpointFile)))).toMap
+  val liveLogDirs: ArrayBuffer[File] = createAndValidateLogDirs(logDirs)
+  @volatile private var recoveryPointCheckpoints = liveLogDirs.map(dir => (dir, new OffsetCheckpointFile(new File(dir, RecoveryPointCheckpointFile)))).toMap
```

Comment:
I think it can be changed to a mutable map since a volatile var map can usually be replaced by a mutable map and vise versa. 

Does Kafka code has a preference between the two? I have chosen to use volatile var here because the code may be simpler. For example, I can simply do `recoveryPointCheckpoints = recoveryPointCheckpoints.filterKeys(file => file.getAbsolutePath != dir)` in `LogManager.handleLogDirFailure(dir: String)`.